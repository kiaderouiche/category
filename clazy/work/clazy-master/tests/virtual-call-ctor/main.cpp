#include <QtCore/QObject>






struct A
{
    A()
    {
        indirection1();
    }

    ~A()
    {
        indirection1();
    }

    void indirection1()
    {
        some_virtual_method();
        some_non_virtual_method();
    }

    virtual void some_virtual_method() = 0;

    void some_non_virtual_method()
    {
    }
};

class B : public QObject {
public:
    B() {
        QObject::connect(this, &QObject::destroyed, this, [this] {
            some_virtual_method();
        });
    }
     virtual void some_virtual_method() = 0;
};
