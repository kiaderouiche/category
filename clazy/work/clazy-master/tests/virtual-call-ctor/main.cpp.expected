virtual-call-ctor/main.cpp:10:5: warning: Calling pure virtual function in CTOR [-Wclazy-virtual-call-ctor]
virtual-call-ctor/main.cpp:12:9: warning: Called here [-Wclazy-virtual-call-ctor]
virtual-call-ctor/main.cpp:15:5: warning: Calling pure virtual function in DTOR [-Wclazy-virtual-call-ctor]
virtual-call-ctor/main.cpp:17:9: warning: Called here [-Wclazy-virtual-call-ctor]
